rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,727 @@
+            schema:
+              $ref: '#/components/schemas/RemovePropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+        '417':
+          description: Namespace not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/NoSuchNamespaceError'
+              example:
+                '{ error: { message: "Namespace does not exist", type: "NoSuchNamespaceException", code: 41701 }'
+        '409':
+          description: Namespace already exists
+          content:
+            applicaton/json:
+              schema:
+                $ref: '#/components/schemas/NamespaceAlreadyExistsError'
+              example:
+                '{ error: { message: "Namespace already exists", type: "AlreadyExistsException", code: 40901 }'
+    post:
+      tags:
+        - Catalog API
+      summary: Overwrite a namespace's properties with a new set of properties
+      description: Set properties on a namespace
+      operationId: setProperties
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SetPropertiesRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                example: '{ data: { success: true }, error: { } }'
+  # TODO - Pagination
+  /v1/namespaces:
+    parameters:
+      - name: parent
+        in: query
+        description: Optional parent namespace under which to list namespaces. When empty, list top-level namespaces.
+        required: false
+        schema:
+          type: string
+    get:
+      tags:
+        - Catalog API
+      summary: List namespaces, optionally providing a parent namespace to list underneaath
+      description:
+        List all namespaces at a certain level, optionally starting from a given parent namespace.
+        For example, if table a.b.t exists, using 'SELECT NAMESPACE IN a' this would translate into
+        `GET /v1/namespaces?parent=a` and must return Namepace.of("a","b").
+      operationId: listNamespaces
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListNamespacesResponse'
+        '401':
+          description: Unauthorized
+  /v1/namespaces/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        required: true
+        schema:
+          type: string
+    post:
+      tags:
+        - Catalog API
+      summary: Create a namespace
+      description: Create a namespace, with an optional set of properties. The server might also add properties.
+      operationId: createNamespace
```

Comment:
I think that `POST /namespaces/{namespace}` to create is inconsistent with `POST /namespaces/{namespace}/tables` to create a table, where the table identifier is in the query. What about `POST /namespaces` instead?

Reply:
That seems reasonable to me. Will update.